### PR TITLE
Adds algo/transitive_closure

### PR DESCRIPTION
**Pull request by bowei**
Transitive closure computes the transitive closure of following resource
dependencies given a root set of resources.

### CODE REVIEW

File: pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+			})
+		}
+		if fake.Mocks.Add(ret) {
+			// We panic here to catch buggy test setup early.
+			panic(fmt.Sprintf("duplicate fake.Mocks.Add(%s)", ret.ID()))
+		}
+		klog.Infof("fake.Mocks.Add(%s)", ret.ID())
+		return ret
+	}
+	addNodeToGraph := func(g *rgraph.Builder, from string, toList []string, opts ...func(*fake.Builder)) {
```
Comment (kl52752) [inline pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go:44]: I don't see `opts` to be ever passed here? Do you really need it?

Comment (bowei) [reply pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go:44]: No, but as test code there isn't much harm having this as an option to future tests...

**Comment by bowei** [comment]
@kl52752 updated please take a look

**Comment by kl52752** [comment]
\lgtm

**Review by swetharepakula — COMMENTED** [review]
LGTM, found one typo

**Pull request merged**